llvm/include/llvm/CodeGen/ScheduleDAG.h

```diff
@@ -788,6 +795,9 @@ class TargetRegisterInfo;
     const_reverse_iterator rbegin() const { return Index2Node.rbegin(); }
     reverse_iterator rend() { return Index2Node.rend(); }
     const_reverse_iterator rend() const { return Index2Node.rend(); }
+
+    /// Returns the SUnit at \p Index, i.e. its Node number.
+    SUnit &getSUnit(int Index) const { return SUnits[Index]; }
```

Comment:
My scheduling usage is rusty but using an `Index` to get a `SUnit` feels inconsistent with the existing APIs. I don't necessarily think it is problematic, just flagging this for consideration.

Reply:
-- For a given value of 'API'. I think this predates the discovery that SUnits is used as a public member all over the place. We have use of mapping by index rather than by MI though.